### PR TITLE
eq_op base implementation

### PR DESCRIPTION
**Pull request by Gerson2102**
### eq_op base implementation
Is not completed. I still have work to do.

### Changes
1. I added a new file for the `eq_op` lint.
2. I created the `comparison` directory for the `eq_op` tests.
3. I added some tests to see the behaviour of the code that I have.
4. I added the lint in the `plugin.rs` file.

### CODE REVIEW

File: crates/cairo-lint-core/src/lints/equal_operand.rs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+pub fn check_redundant_comparison(db: &dyn SyntaxGroup, expr: &ExprBinary, diagnostics: &mut Vec<PluginDiagnostic>) {
+    let lhs = expr.lhs(db);
+    let rhs = expr.rhs(db);
+
+    let is_function_or_method_call = |expr: &Expr| match expr {
+        Expr::FunctionCall(_) => true,
+        Expr::StructCtorCall(_) => true,
+        _ => false,
+    };
```
Comment (lauchaves) [inline crates/cairo-lint-core/src/lints/equal_operand.rs:17]: hey @Gerson2102 is this

`pub fn is_function_or_method_call(db: &dyn SyntaxGroup, expr: &Expr) -> bool {
    let syntax_node = expr.as_syntax_node();
    match syntax_node.kind(db) {
        SyntaxKind::ExprFunctionCall | SyntaxKind::ExprStructCtorCall => true,
        _ => false,
    }
}
` 

the same approach as u have there? maybe checking it from `SyntaxKind` does the trick 🤔  

 `use cairo_lang_syntax::node::kind::SyntaxKind;`

File: crates/cairo-lint-core/src/plugin.rs [hunk 2]
```diff
@@ -82,6 +84,12 @@ impl AnalyzerPlugin for CairoLint {
                         &mut diags,
                     ),
                     SyntaxKind::StatementBreak => breaks::check_break(db.upcast(), node, &mut diags),
+                    SyntaxKind::ExprBinary => {
+                        // Convertir el nodo a `ExprBinary` utilizando `match`
+                        if let AstExpr::Binary(expr_binary) = AstExpr::from_syntax_node(db.upcast(), node) {
+                            equal_operand::check_redundant_comparison(db.upcast(), &expr_binary, &mut diags);
+                        }
+                    }
```
Comment (0xLucqs) [inline crates/cairo-lint-core/src/plugin.rs:92]: I don't think you should check using the syntax nodes, add the check in the iteration above, similarly to [that](https://github.com/keep-starknet-strange/cairo-lint/blob/e6a73108dc281b78533088c9e16700e3d3ca8230/crates/cairo-lint-core/src/plugin.rs#L51-L56). Then you'll need to check the `cairo_lang_semantic::Expr::If`. check [this](https://github.com/keep-starknet-strange/cairo-lint/blob/main/crates/cairo-lint-core/src/lints/loops.rs) and [this](https://github.com/keep-starknet-strange/cairo-lint/blob/main/crates/cairo-lint-core/src/lints/single_match.rs) for inspiration. Let me know if that helps or not

Comment (Gerson2102) [reply crates/cairo-lint-core/src/plugin.rs:92]: I see. It seems that I have to change almost everything if I'm not wrong.

**Review by 0xLucqs — CHANGES_REQUESTED** [review]

**Comment by Gerson2102** [comment]
Is weird that my commit says that I changed the bool_comparision tests when I did not. I think I have to be on a commit before right? @0xLucqs

**Comment by 0xLucqs** [comment]
any news ?

**Comment by Gerson2102** [comment]
I have not been spending time in it. I dont know if Im gonna be able to solve it.

**Comment by 0xLucqs** [comment]
Ok i'll be able to provide more help next week

**Comment by stevencartavia** [comment]
I would like to take over this one

**Comment by Gerson2102** [comment]
I wanted to solve other issues that I had that I knew that are easier than this one xD

**Pull request closed without merging**